server/applicationset/applicationset.go

```diff
@@ -274,7 +275,7 @@ func (s *Server) generateApplicationSetApps(ctx context.Context, logEntry *log.E
 		return nil, fmt.Errorf("error creating ArgoCDService: %w", err)
 	}
 
-	appSetGenerators := generators.GetGenerators(ctx, s.client, s.k8sClient, namespace, argoCDService, s.dynamicClient, scmConfig)
+	appSetGenerators := generators.GetGenerators(ctx, s.client, s.k8sClient, namespace, argoCDService, s.dynamicClient, scmConfig, matrixConfig)
```

Comment:
I have resolved all conflicts but I don't know how to get the config in this case. This method is used in `argocd appset create --dry-run` and `argocd appset generate`. I think not setting a limit here makes sense.